File: source/26_Remove Duplicates from Sorted Array.c
int removeDuplicates(int* nums, int numsSize) {
    if(numsSize<2)
        return numsSize;
    if(numsSize==2)
        return nums[0]==nums[1]?1:2;//[a,a] must be considered
    int index=1;
    for(int i=1;i<numsSize;i++){
        if(*(nums+i)!=*(nums+i-1)){
            nums[index]=*(nums+i);
            index +=1;
        }
    }
    return index;
}

File: source/26_Remove Duplicates from Sorted Array.java
public class Solution {
    public int removeDuplicates(int[] nums) {
        if(nums.length == 1)
            return 1;
        if(nums.length == 2)
            return nums[0] == nums[1] ? 1 : 2;
        int size = 1;
        for(int i = 1; i < nums.length; i++) {
            if(nums[i] != nums[i - 1])
                nums[size++] = nums[i];
        }
        return size;
    }
}
